website/cue/reference/components/sources/socket.cue

```diff
@@ -88,6 +88,7 @@ components: sources: socket: {
 		max_length: {
 			common:      true
 			description: "The maximum bytes size of incoming messages before they are discarded."
+			relevant_when: "mode = `unix_datagram`"
```

Comment:
Why did we only keep max length for unix datagrams?

Reply:
For TCP and Udp, this must now be controlled by setting the `max_length` on the `framing` config.

Eventually, I'd also like to remove it for Unix since this option does control the `max_length` in an indirect way (by setting the receive buffer size) or at least rename it to match that behavior better?

Unfortunately there is no `UnixFramed` like tokio's `UdpFramed`. `UdpFramed` sets the receive buffer size to UDP's MTU: https://github.com/tokio-rs/tokio/blob/7ce8f05cff4bea1379aea07fcb14f610cbc17da4/tokio-util/src/udp/frame.rs#L50. I'm wondering - does there even exist a MTU for Unix sockets?

Reply:
[Seems](https://stackoverflow.com/questions/4729315/what-is-the-max-size-of-af-unix-datagram-message-in-linux) like size is limited by the ability of the kernel to find contiguous memory to stick it in?